Code/autopkglib/github/__init__.py

```diff
@@ -111,129 +182,26 @@ def call_api(
                 assets)."""
 
         # Compose the URL
-        url = BASE_URL + endpoint
+        self.env["url"] = BASE_URL + endpoint
         if query:
-            url += "?" + query
-
-        try:
-            # Compose the curl command
-            curl_path = curl_cmd()
-            if not curl_path:
-                return (None, None)
-            cmd = [
-                curl_path,
-                "--location",
-                "--silent",
-                "--show-error",
-                "--fail",
-                "--dump-header",
-                "-",
-            ]
-            cmd.extend(["-X", method])
-            cmd.extend(["--header", "%s: %s" % ("User-Agent", "AutoPkg")])
-            cmd.extend(["--header", "%s: %s" % ("Accept", accept)])
-
-            # Pass the GitHub token as a header
-            if self.token:
-                cmd.extend(
-                    ["--header", "%s: %s" % ("Authorization", "token %s" % self.token)]
-                )
+            self.env["url"] += "?" + query
 
-            # Additional headers if defined
-            if headers:
-                for header, value in headers.items():
-                    cmd.extend(["--header", "%s: %s" % (header, value)])
-
-            # Set the data header if defined
-            if data:
-                data = json.dumps(data)
-                cmd.extend(["-d", data, "--header", "Content-Type: application/json"])
-
-            # Final argument to curl is the URL
-            cmd.append(url)
-
-            # Start the curl process
-            proc = subprocess.Popen(
-                cmd,
-                shell=False,
-                bufsize=1,
-                stdout=subprocess.PIPE,
-                stderr=subprocess.PIPE,
-            )
+        temp_content = tempfile.NamedTemporaryFile().name
```

Comment:
We are now saving the data output to the temporary file so headers cant be parsed from curl output via URLGetter.